File: src/main/java/me/dyzjct/kura/notification/NotificationManager.java
package me.dyzjct.kura.notification;

import java.util.concurrent.LinkedBlockingQueue;
import net.minecraft.client.Minecraft;

public class NotificationManager {
    public static final LinkedBlockingQueue<Notification> pendingNotifications = new LinkedBlockingQueue<>();
    private static Notification currentNotification = null;

    public static void show(Notification notification) {
        pendingNotifications.add(notification);
    }

    public static void update() {
        if (currentNotification != null && !currentNotification.isShown()) {
            currentNotification = null;
        }
        if (currentNotification == null && !pendingNotifications.isEmpty()) {
            currentNotification = pendingNotifications.poll();
            currentNotification.show();
        }
    }

    public static void render() {
        try {
            int divider = Minecraft.getMinecraft().gameSettings.guiScale;
            int Dwidth = Minecraft.getMinecraft().displayWidth / divider;
            int Dheight = Minecraft.getMinecraft().displayHeight / divider;
            NotificationManager.update();
            if (currentNotification != null) {
                currentNotification.render(Dwidth, Dheight);
            }
        }catch (Exception ignored){}
    }

}

File: src/main/java/me/windyteam/kura/notifications/NotificationManager.kt
package me.windyteam.kura.notifications

import net.minecraftforge.common.MinecraftForge
import java.util.concurrent.CopyOnWriteArrayList

class NotificationManager {

    val notifications = CopyOnWriteArrayList<Notification>()

    init {
        MinecraftForge.EVENT_BUS.register(this)
    }

    fun addNotification(notification: Notification) = notifications.add(notification)

}